src/libs/Network/SequentialQueue.js

```diff
@@ -95,13 +112,9 @@ function push(request) {
         return;
     }
 
-    // If the queue is running this request will run once it has finished processing the current batch
-    if (isSequentialQueueRunning) {
-        isReadyPromise.then(flush);
-        return;
+    if (!isSequentialQueueRunning) {
+        flush();
     }
-
-    flush();
```

Comment:
I think I'm going to revert these changes because we need the isReadyPromise for `waitForIdle()`. API.read requests don't happen until the SequentialQueue is idle.

Reply:
I'm not sure what you mean, but read requests shouldn't be in the sequential queue, once we start the sequential queue we should process all the requests in it before making any other requests, so there shouldn't be any read requests happening in between the sequential queues requests.

Reply:
> I'm not sure what you mean

I'll try to clarify, I could have been more clear. In API.read we wait for the sequential queue to be idle before sending the request. API.read can be called while there are write requests in the sequential queue. API.read uses the waitForIdle function which requires the `isReadyPromise` which is currently undefined, so I need to add it again. All of this was added since you last worked on this PR so it simply needs to be updated now that I've merged in main.